src/Servers/Kestrel/Core/src/Middleware/Internal/DuplexPipeStreamAdapter.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    /// <summary>
+    /// A helper for wrapping a Stream decorator from an <see cref="IDuplexPipe"/>.
+    /// </summary>
+    /// <typeparam name="TStream"></typeparam>
+    internal class DuplexPipeStreamAdapter<TStream> : DuplexPipeStream, IDuplexPipe where TStream : Stream
```

Comment:
Are we planning to ship something like this in 3.1 in corefx?

Reply:
I've informally spoken to a few people about it.